sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -748,7 +775,27 @@ export class ServiceBusAtomManagementClient extends ServiceClient {
   ): Promise<HttpOperationResponse> {
     const webResource: WebResource = new WebResource(this.getUrl(name), "GET");
 
-    return executeAtomXmlOperation(this, webResource, serializer);
+    return new Promise<HttpOperationResponse>(async (resolve, reject) => {
```

Comment:
There shouldn't be a need to wrap this in a new Promise since `getResource` is already an async function. You should be able to do refactor like this:

```ts
// don't need try/catch around this because we want the error to be thrown anyway
const response = await executeAtomXmlOperation(this, webResource, serializer);
if (
  response.parsedBody == undefined ||
  (Array.isArray(response.parsedBody) && response.parsedBody.length == 0)
) {
  const err = new RestError(
    `The messaging entity "${name}" being requested cannot be found.`,
    "404",
    404,
    stripRequest(webResource),
    stripResponse(response)
  );
  throw err;
}
return response;
```

Reply:
Updated